public/Add-DbaReplArticle.ps1

```diff
@@ -0,0 +1,179 @@
+        [PSCredential]$SqlCredential,
+        [parameter(Mandatory)]
+        [String]$Database,
+        [parameter(Mandatory)]
+        [String]$Publication,
+        [String]$Schema = 'dbo',
+        [parameter(Mandatory)]
+        [String]$Name,
+        [String]$Filter,
+        [PSObject]$CreationScriptOptions,
+        [Switch]$EnableException
+    )
+    process {
+
+        # Check that $CreationScriptOptions is a valid object
+        if ($CreationScriptOptions -and ($CreationScriptOptions -isnot [Microsoft.SqlServer.Replication.CreationScriptOptions])) {
+            Stop-Function -Message "CreationScriptOptions should be the right type. Use New-DbaReplCreationScriptOptions to create the object" -ErrorRecord $_ -Target $instance -Continue
+        }
+
+        foreach ($instance in $SqlInstance) {
+            try {
+                $replServer = Get-DbaReplServer -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+            Write-Message -Level Verbose -Message "Adding article $Name to publication $Publication on $instance"
+
+            try {
+                if ($PSCmdlet.ShouldProcess($instance, "Adding an article to $Publication")) {
+
+                    $pub = Get-DbaReplPublication -SqlInstance $instance -SqlCredential $SqlCredential -Name $Publication
+
+                    $articleOptions = New-Object Microsoft.SqlServer.Replication.ArticleOptions
+
+                    if ($pub.Type -in ('Transactional', 'Snapshot')) {
+                        $article = New-Object Microsoft.SqlServer.Replication.TransArticle
+                        $article.Type = $ArticleOptions::LogBased
+                    } elseif ($pub.Type -eq 'Merge') {
+                        $article = New-Object Microsoft.SqlServer.Replication.MergeArticle
+                        $article.Type = $ArticleOptions::TableBased
+                    }
+
+                    $article.ConnectionContext = $replServer.ConnectionContext
+                    $article.Name = $Name
+                    $article.DatabaseName = $Database
+                    $article.SourceObjectName = $Name
+                    $article.SourceObjectOwner = $Schema
+                    $article.PublicationName = $Publication
+
+                    if ($CreationScriptOptions) {
+                        $article.SchemaOption = $CreationScriptOptions
+                    }
+
+                    if ($Filter) {
+                        if ($Filter -like 'WHERE*') {
+                            Stop-Function -Message "Filter should not include the word 'WHERE'" -ErrorRecord $_ -Target $instance -Continue
+                        }
+                        $article.FilterClause = $Filter
+                    }
+
+                    if (-not ($article.IsExistingObject)) {
+                        $article.Create()
```

Comment:
It would be cleaner if the try/catch was only focused on this line instead of the whole code block (130-173)